packages/common/src/dynamicLoading/dynamic-font-loading.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * The following imports are always present when react native is installed
+ * in the future, more explicit apis will be exposed by the core, including typings
+ * */
+// @ts-expect-error missing types
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { getAssetByID } from '@react-native/assets-registry/registry';
+import type { Image } from 'react-native';
+// @ts-expect-error missing types
+import resolveAssetSource from 'react-native/Libraries/Image/resolveAssetSource';
+import type { FontSource } from './types';
+import { getErrorCallback } from './dynamic-loading-setting';
+
+const loadPromises: { [fontSource: string]: Promise<void> } = {};
+
+const loadFontAsync = async (fontFamily: string, fontSource: FontSource): Promise<void> => {
+  const expoModules = globalThis?.expo?.modules;
+  if (!expoModules) {
+    throw new Error('Expo is not available. Dynamic font loading is not available.');
+  }
+  if (Object.prototype.hasOwnProperty.call(loadPromises, fontFamily)) {
```

Comment:
Can this be simplified to loadPromises[fontFamily]?

Reply:
done. I had it differently but biome didn't like the approach and suggested `Object.prototype.hasOwnProperty` 😄 . This is better.